core/src/test/java/com/google/cloud/sql/core/LazyRefreshStrategyTest.java

```diff
@@ -106,13 +103,19 @@ public void testCloudSqlRefreshesExpiredData() throws Exception {
     assertThat(refreshCount.get()).isEqualTo(1);
     assertThat(d).isSameInstanceAs(initialData);
 
+    waitForExpiration(initialData);
+
+    assertThat(r.getConnectionInfo(TEST_TIMEOUT_MS)).isSameInstanceAs(data);
+    assertThat(refreshCount.get()).isEqualTo(2);
+  }
+
+  private static void waitForExpiration(ExampleData initialData) throws InterruptedException {
     // Wait for the instance to expire
     while (Instant.now().isBefore(initialData.getExpiration())) {
       Thread.sleep(10);
     }
-
-    assertThat(r.getConnectionInfo(TEST_TIMEOUT_MS)).isSameInstanceAs(data);
-    assertThat(refreshCount.get()).isEqualTo(2);
+    // Introduce a little more delay to allow for Java 8
```

Comment:
Could we elaborate on the comment here? Why does JAva 8 need more time?

Reply:
Alternatively, could we make the sleep above be for 100ms instead?

Reply:
I updated this to make the comment a little more explicit.